src/tiled/changemapobject.h

```diff
@@ -87,5 +87,27 @@ class ChangeMapObjectCells : public QUndoCommand
     QVector<MapObjectCell> mChanges;
 };
 
+class ChangeMapObjectsTile : public QUndoCommand
+{
+public:
+    ChangeMapObjectsTile(MapDocument *mapDocument,
+                       const QList<MapObject *> &mapObjects,
+                       Tile *tile);
+
+    void undo() override { restoreTiles(); }
+    void redo() override { changeTiles(); }
+
+private:
+    void changeTiles();
+    void restoreTiles();
+
+    MapDocument *mMapDocument;
+    const QList<MapObject *> mMapObjects;
+    Tile * const mTile;
+    QList<Tile *> mOldTiles;
+    QList<QSizeF> mOldSize;
```

Comment:
Please use a `QVector<QSizeF>`,  because a `QSizeF` does not fit in `sizeof(void*)` (making QList very inefficient for this type, since it will allocate each size on the heap and store a pointer to it). Actually you could use a QVector for all containers, except for `mMapObjects` (since it would require changing too much other code to avoid copying from QList, and there is no reason not to use a QList for storing pointers.